docs/design-notes/system_keyspace.md

```diff
@@ -2,119 +2,82 @@
 
 This section describes layouts and usage of system.* tables.
 
-## The system.large\_* tables
+## system.large\_data
 
-Scylla performs better if partitions, rows, or cells are not too
-large. To help diagnose cases where these grow too large, scylla keeps
-3 tables that record large partitions, rows, and cells, respectively.
+Scylla performs better if partitions, rows, and individual cells are
+not too large. To help diagnose cases where these grow too large,
+scylla keeps a table that records large partitions, rows, and cells.
 
-The meaning of an entry in each of these tables is similar. It means
-that there is a particular sstable with a large partition, row, or
-cell. In particular, this implies that:
+Each entry means that there is a particular sstable with a large
+partition, row, or cell.  In particular, this implies that:
 
 * There is no entry until compaction aggregates enough data in a
   single sstable.
 * The entry stays around until the sstable is deleted.
 
 In addition, the entries also have a TTL of 30 days.
 
-## system.large\_partitions
-
-Large partition table can be used to trace largest partitions in a cluster.
-
 Schema:
 ~~~
-CREATE TABLE system.large_partitions (
+CREATE TABLE system.large_data (
     keyspace_name text,
     table_name text,
     sstable_name text,
-    partition_size bigint,
+    ratio float,
+    metric text,
     partition_key text,
+    clustering_key text,
+    column_name text,
+    value bigint,
     compaction_time timestamp,
-    PRIMARY KEY ((keyspace_name, table_name), sstable_name, partition_size, partition_key)
-) WITH CLUSTERING ORDER BY (sstable_name ASC, partition_size DESC, partition_key ASC);
+    PRIMARY KEY ((keyspace_name, table_name, sstable_name), ratio, partition_key, clustering_key, column_name)
+) WITH CLUSTERING ORDER BY (ratio DESC, partition_key ASC, clustering_key ASC, column_name ASC);
 ~~~
 
+The ratio (think "badness") column is there to provide a way to
+meaningfully sort heterogeneous data -- no useful ordering can be
+defined between, say, "12mb row" and "2mb cell" as absolute values,
+but (if the thresholds are 10mb and 1mb respectively) 2x is clearly
+worse than 1.2x, so the cell will be shown before the row.
+
 ### Example usage
 
-#### Extracting large partitions info
+(Assume `ALLOW FILTERING` throughout).
+
+#### Extracting large data info
 ~~~
-SELECT * FROM system.large_partitions;
+SELECT * FROM system.large_data;
 ~~~
```

Comment:
We can't do this. If someone scripted large_partitions, their scripts will break.